src/Parsers/ASTWithAliasPending.cpp

```diff
@@ -0,0 +1,33 @@
+#include <Parsers/ASTWithAliasPending.h>
+#include <Parsers/ASTWithAlias.h>
+#include <Parsers/ASTQueryParameter.h>
+#include <IO/Operators.h>
+
+namespace DB
+{
+
+ASTWithAliasPending::ASTWithAliasPending(ASTPtr wrapped_, ASTPtr param_)
```

Comment:
I like this feature, and we need it!

I'm not sure if it is the simplest implementation.
Maybe we can allow ParserIdentifier to product ASTQueryParameter?

Reply:
It's funny, I just repeated what @tavplubix said earlier :)